File: probs/usaco/phrase/phrase.cpp
// Solution by: Albert Ye
// AlphaStar 2020 Smmmer Camp: CC31AB
// Problem: feb06-phrase
#include <bits/stdc++.h>

using namespace std;

int n, m;
string book[10005];
int main()
{
	cin >> m >> n;
	getline(cin, book[0]); // deal with newline
	for(int i = 0; i < m; i++)
	{
		getline(cin, book[i]); // input whole line
	}
	// sort books s.t. we can binary search on them
	sort(book, book+m);
	int ans = 0;
	for(int i = 0; i < n; i++)
	{
		string s;
		getline(cin, s);
		string t = *(lower_bound(book, book+m, s)); // the phrase in the book
		// lexicographically following the phrase.
		t.resize(s.size()); // we only care about the first |s| characters of each string when finding if s is a prefix.
		if(s == t) // if prefix matches
			ans++;
	}
	// O(n lg m) solution, as intended
	cout << ans << endl;
}

File: probs/usaco/phrase/phrase.py
"""
Solution by Albert Ye
AlphaStar 2020 Summer Camp: CC31AB
Problem: feb06-phrase
"""
# for binary search
import bisect
m, n = input().split(' ')
n = int(n)
m = int(m)
# phrasebook
book = []
for i in range(m) :
    book.append(input())
# sort phrasebook
book.sort()
# number of valid prefixes
ans = 0
for i in range(n) :
    s = input()
    # similar to c++'s lower_bound()
    x = bisect.bisect_left(book,s)
    # check if its index is out of bounds, in which case
    # it can't be a prefix
    if x >= len(book) :
        continue
    t = book[x]
    # match all characters of s, t and see if it's prefix
    is_prefix = True
    # check if characters match
    for j in range(min(len(t), len(s))) :
        if s[j] != t[j] :
            is_prefix = False
    # if they match, add to the answer
    if is_prefix :
        ans+=1
print(ans)
